.vscode/launch.template.json

```diff
@@ -0,0 +1,59 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+      {
+        "type": "node",
+        "name": "Attach",
+        "port": 9229,
+        "request": "attach",
+        "skipFiles": ["<node_internals>/**"]
+      },
+      {
+        "type": "node",
+        "request": "launch",
+        "name": "Beacon",
+        "skipFiles": ["<node_internals>/**"],
+        "smartStep": true,
+        "program": "${workspaceFolder}/packages/cli/bin/lodestar.js",
+        "args": ["beacon"],
+        "console": "integratedTerminal"
+      },
+      {
+        "type": "node",
+        "request": "launch",
+        "name": "Dev",
+        "skipFiles": ["<node_internals>/**"],
+        "smartStep": true,
+        "program": "${workspaceFolder}/packages/cli/bin/lodestar.js",
+        "args": ["dev"],
+        "console": "integratedTerminal"
+      },
+      {
+        "name": "Test Current File",
+        "type": "node",
+        "request": "launch",
+        "program": "${workspaceFolder}/node_modules/.bin/vitest",
+        "args": ["--run", "${file}", "-t", "${input:testName}"],
+        "cwd": "${workspaceFolder}/${input:packageName}",
+        "console": "integratedTerminal",
+        "skipFiles": ["<node_internals>/**"]
+      }
+    ],
+    "inputs": [
```

Comment:
Would it be possible to do another input block to extract the test type so it applies the correct vitest config?  

```json
        "id": "testType",
        "type": "command",
        "command": "extension.commandvariable.transform",
        "args": {
          "text": "${relativeFileDirname}",
          "find": "^(test/[^/]+).*",
          "replace": "$1"
        }
```

That way in the test config we could do something like this
```json
            "program": "${workspaceFolder}/node_modules/.bin/vitest",
            "args": [
                "--run",
                "--config",
                "vitest.${input:testType}.config.ts",
                "-t",
                "${input:testName}",
                "${file}"
            ],
```

Reply:
in what scenario do we have to specify a different config? seems like the default config works for debugging

Reply:
There are some additional things that are in the separate configs for spec and e2e tests.  Guessing we may add more stuff over time so will be nice to have the correct config passed for the testing environment.

https://github.com/ChainSafe/lodestar/blob/b6bb66057b59ec32d3c5c7dd1cc70281b4a5ed82/vitest.base.spec.config.ts#L4

https://github.com/ChainSafe/lodestar/blob/b6bb66057b59ec32d3c5c7dd1cc70281b4a5ed82/vitest.base.e2e.config.ts#L4

Reply:
I don't see how those are relevant for debugging, I would rather keep that out for now, having 2 prompts seems a bit annoying, especially if as of right now you would never change the spec config to use.

I think we rather wanna set up the debug launch config in a way it's optimized for debugging.

Reply:
There should not be a second prompt.  It would pull the value from the filepath like the packageName is handled

Reply:
>  It would pull the value from the filepath like the packageName is handled

that's a great idea, we should add that

Reply:
@matthewkeil feel free to adapt the existing one or share some related snippets!

Reply:
Got it working fine when the file exists but I tried unsuccessfully for a couple hours to figure out a way to not include the flag if the file doesn't exist.  Its starting to not be worth it.... approving as is.